### PR TITLE
chore: update on product unavailable ui

### LINKED ISSUE #206: Unavailable product
Unavailable product should be displayed consistently to other integrations:

* information that product is out of stock
* disabled quantity selection
* disabled 'add to cart' button

┆Issue is synchronized with this [Jira Zadanie](https://vsf.atlassian.net/browse/SH-65) by [Unito](https://www.unito.io)
┆Attachments: <a href="https://vsf.atlassian.net/rest/api/2/attachment/content/11140">Screen Shot 2022-02-02 at 12.07.00.png</a>

### PR DESCRIPTION
**Pull request by jeffpdotone**
Fixed #206 

## Description
This a maintenance work on making the Shopify storefront adhere to the standard design of the storefront UI

## How Has This Been Tested?
Tested in local with post check on deployed version

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/pages/Product.vue [hunk 1]
```diff
@@ -162,58 +133,66 @@
...
+              >
+                <template v-if="property.name === 'Category'" #value>
+                  <SfButton class="product__property__button sf-button--text">
+                    {{ property.value }}
+                  </SfButton>
+                </template>
+              </SfProperty>
+            </SfTab>
+            <SfTab
+              title="Additional Information"
```
Comment (Baroshem) [inline packages/theme/pages/Product.vue:173]: Maybe add i18n here?

File: packages/theme/pages/Product.vue [hunk 2]
```diff
@@ -395,8 +375,11 @@ export default {
 
     onSSR(async () => {
       await search({ slug, selectedOptions: configuration.value }).then(() => {
-        if (productTitle.value === 'Product\'s name') {
-          return context.root.error({ statusCode: 404, message: 'This product could not be found' });
+        if (productTitle.value === "Product's name") {
```
Comment (Baroshem) [inline packages/theme/pages/Product.vue:378]: Could you provide some comment or a variable explaining why this case should be redirected to 404? I needed a while to understand why it is an issue :)

File: packages/theme/pages/Product.vue [hunk 3]
```diff
@@ -162,58 +133,66 @@
             <template #add-to-cart-btn>
               <SfButton
                 class="sf-add-to-cart__button"
-                :disabled="loading"
-                @click="addingToCart({ product, quantity: parseInt(qty), customQuery: [{key: 'CustomAttrKey', value: 'CustomAttrValue'}]})"
+                :disabled="loading || !productGetters.getStockStatus(product)"
+                @click="
+                  addingToCart({
```
Comment (Baroshem) [inline packages/theme/pages/Product.vue:138]: I am not sure about this formatting. Could you check once again if it is correct?

File: packages/theme/pages/Product.vue [hunk 4]
```diff
@@ -144,16 +101,30 @@
...
-                  "
+                  @click="(atttLbl = o), updateFilter({ [atttLbl]: attribs })"
                 />
               </div>
             </template>
           </div>
+
+          <SfAlert
+            v-if="!productGetters.getStockStatus(product)"
+            message="Out of Stock"
```
Comment (Baroshem) [inline packages/theme/pages/Product.vue:112]: Here i18n would be useful as well

File: packages/theme/pages/Product.vue [hunk 5]
```diff
@@ -124,14 +79,16 @@
                 </SfSelectOption>
               </SfSelect>
               <div
-                v-else
+                v-else-if="o.toLowerCase() === 'color'"
```
Comment (Baroshem) [inline packages/theme/pages/Product.vue:82]: Are you sure it should be `o`, not `option`? o seems to be an index of an array. Also in a previous step you are checking of something does not equal color. Do you explicilty have to check in this step for someting that exactly equals color?

**Review by Baroshem — APPROVED** [review]
Overall approve, some small recommendations and questions

File: packages/theme/pages/Product.vue (hunk 1 above)
Comment (Mateuszpietrusinski) [reply packages/theme/pages/Product.vue:173]: Yes, its a must

File: packages/theme/pages/Product.vue (hunk 2 above)
Comment (jeffpdotone) [reply packages/theme/pages/Product.vue]: Added comments to it and modified to make it clearer that product title is determining if the the product is existing or not

File: packages/theme/pages/Product.vue (hunk 3 above)
Comment (jeffpdotone) [reply packages/theme/pages/Product.vue:138]: This seems to be across as this formatting but I'll put in notes to improve this.

File: packages/theme/pages/Product.vue (hunk 5 above)
Comment (jeffpdotone) [reply packages/theme/pages/Product.vue]: This is actually the key to the options. it's using ... in ... loop method.

**Pull request merged**